src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -115,12 +126,42 @@ protected Endpoint getEndpoint() {
         return Endpoint.INTERNALUSERS;
     }
 
+    @Override
+    protected void handleGet(final RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+        final String resourcename = request.param("name");
+
+        SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+
+        String includeIfType = request.param("filterBy", "any");
+
+        if (includeIfType != "any") {
+            try {
+                userService.includeAccountsIfType(configuration, AccountType.valueOf(includeIfType.toUpperCase()));
+            } catch (UserServiceException ex) {
+                badRequestResponse(channel, ex.getMessage());
+            }
```

Comment:
Lets avoid using exceptions as flow of control, the only invalid case is when the account type isn't known AFAIK, maybe we could change the method to accept `null` or `Optional.none()` and not have to special case.

Reply:
Should it return everything or nothing if that is case?

Reply:
If nothing is specified, you can return all. If an invalid account type is provided just return nothing. I am not opposed to throwing the exception personally, but the benefit of applying the filter always is that it would be more future proof. 

I.e. you filter for "purple," you get nothing because we don't have purple accounts. But in the future we could and the call would already work. 

Empty filter = everything.